network/fakepulsar/fakepulsar.go

```diff
@@ -35,20 +36,23 @@ import (
 type FakePulsar struct {
 	onPulse   network.PulseHandler
 	stop      chan bool
-	timeoutMs int32 // ms
-	pulse     int
+	timeoutMs int32 // mspulse     int
 
-	mutex   sync.RWMutex
-	running bool
+	mutex          sync.RWMutex
+	running        bool
+	firstPulseTime int64
+	pulseNum       int64
 }
 
 // NewFakePulsar creates and returns a new FakePulsar.
 func NewFakePulsar(callback network.PulseHandler, timeoutMs int32) *FakePulsar {
 	return &FakePulsar{
-		onPulse:   callback,
-		timeoutMs: timeoutMs,
-		stop:      make(chan bool, 1),
-		running:   false,
+		onPulse:        callback,
+		timeoutMs:      timeoutMs,
+		stop:           make(chan bool, 1),
+		running:        false,
+		firstPulseTime: time.Now().UTC().Unix(),
```

Comment:
set it to zero in constructor and check for zero where it is used (to ensure that SetPulseData was called)

Reply:
fixed